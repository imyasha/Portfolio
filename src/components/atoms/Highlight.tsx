import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Accent = styled.span`
  color: #e94e1b;
`;

interface IProps {
  children: React.ReactNode;
}

const Highlight = ({ children }: IProps) => <Accent>{children}</Accent>;

Highlight.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Highlight;
