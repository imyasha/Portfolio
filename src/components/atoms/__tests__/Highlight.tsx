import React from "react"
import renderer from "react-test-renderer"

import Highlight from "../Highlight"

it("renders correctly", () => {
  const tree = renderer.create(<Highlight>Highlighted text</Highlight>).toJSON()
  expect(tree).toMatchSnapshot()
})
