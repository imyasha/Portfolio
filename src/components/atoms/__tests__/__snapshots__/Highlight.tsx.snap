// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: #e94e1b;
}

<span
  class="c0"
>
  Highlighted text
</span>
`;
